src/main/java/com/feedzai/commons/sql/abstraction/dml/Union.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2019 Feedzai
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.feedzai.commons.sql.abstraction.dml;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.LinkedList;
+import java.util.List;
+
+/**
+ * The union clause.
+ *
+ * @author Francisco Santos (francisco.santos@feedzai.com)
+ * @since 2.3.1
+ */
+public class Union extends Expression {
```

Comment:
With or Union?  
You should have an independent pull request for Union before implementing With.

Reply:
There was a problem with the rebase. Going to fix it.